### PR TITLE
[ads] Unify iOS kBraveNTPBrandedWallpaper

### LINKED ISSUE #8701: [ads] Unify iOS kBraveNTPBrandedWallpaper
Corresponding changes to https://github.com/brave/brave-browser/issues/35522

### PR DESCRIPTION
**Pull request by tmancey**
## Summary of Changes

This pull request fixes #8701

Requires https://github.com/brave/brave-core/pull/21751

## Submitter Checklist:

- [ ] *Unit Tests* are updated to cover new or changed functionality
- [ ] User-facing strings use `NSLocalizableString()`
- [ ] New or updated UI has been tested across:
  - [ ] Light & dark mode
  - [ ] Different size classes (iPhone, landscape, iPad)
  - [ ] Different dynamic type sizes

## Test Plan:

- Fresh install
- Launch browser
- Wait a few minutes for the seed to be fetched (or you can background/foreground the browser)
- Open *Settings > Version #.# (#.#.#.#) > View all version info > Active Variations** 
- Check if `Active Variations:` list contains the enabled studies for iOS. 
- If enabled studies for iOS are not listed in `Active Variations:` then terminate and start browser. Make sure that `Active Variations:` list contains the enabled studies for iOS.
- Open new tabs until an New Tab Takeover ad is shown
    - EXPECTED RESULT: New Tab Takeover ad should be shown on the nth `BraveNTPBrandedWallpaper`/`initial_count_to_branded_wallpaper` Griffin feature param
- Open new tabs until an New Tab Takeover ad is shown
    - EXPECTED RESULT: New Tab Takeover ad should be shown on the nth `BraveNTPBrandedWallpaper`/`count_to_branded_wallpaper` Griffin feature param

We should also test the default values if the Griffin feature does not exist. Default value is 2 for `BraveNTPBrandedWallpaper`/`initial_count_to_branded_wallpaper` and 3 for `BraveNTPBrandedWallpaper`/`count_to_branded_wallpaper`.

And that the position is persisted across browser restarts.

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Comment by tmancey** [comment]
Tests are failing because we first need to merge brave-core changes with https://github.com/brave/brave-core/pull/21751

**Review by aseren — APPROVED** [review]

**Review by Brandon-T — APPROVED** [review]

**Review by Brandon-T — APPROVED** [review]

File: App/iOS/Delegates/AppState.swift [hunk 1]
```diff
@@ -178,7 +178,9 @@ public class AppState {
       }
     }
     switches.append(.init(key: .rewardsFlags, value: BraveRewards.Configuration.current().flags))
-    
+    switches.append(.init(key: .init(rawValue: "disable-variations-seed-fetch-throttling"), value: "true"))
+    switches.append(.init(key: .init(rawValue: "fake-variations-channel"), value: "stable"))
```
Comment (kylehickinson) [inline App/iOS/Delegates/AppState.swift]: Delete

**Review by kylehickinson — CHANGES_REQUESTED** [review]

File: App/iOS/Delegates/AppState.swift (hunk 1 above)
Comment (tmancey) [reply App/iOS/Delegates/AppState.swift]: @Brandon-T

Comment (aseren) [reply App/iOS/Delegates/AppState.swift]: removed `disable-variations-seed-fetch-throttling` and `fake-variations-channel` switches

**Review by kylehickinson — APPROVED** [review]

**Pull request merged**